docs/src/pages/components/data-grid/rendering/AntDesignGrid.js

```diff
@@ -123,14 +123,280 @@ CustomPagination.propTypes = {
+           */
+          height: PropTypes.number.isRequired,
+          /**
+           * The width of a container or HTMLElement.
+           */
+          width: PropTypes.number.isRequired,
+        }).isRequired,
+      }),
+    ]).isRequired,
+    density: PropTypes.shape({
+      headerHeight: PropTypes.number.isRequired,
+      rowHeight: PropTypes.number.isRequired,
+      value: PropTypes.oneOf(['comfortable', 'compact', 'standard']).isRequired,
+    }).isRequired,
+    filter: PropTypes.shape({
+      items: PropTypes.arrayOf(
+        PropTypes.shape({
+          columnField: PropTypes.string,
+          id: PropTypes.number,
+          operatorValue: PropTypes.string,
+          value: PropTypes.string,
+        }),
+      ).isRequired,
+      linkOperator: PropTypes.oneOf(['and', 'or']),
+    }).isRequired,
+    isScrolling: PropTypes.bool.isRequired,
+    keyboard: PropTypes.shape({
+      cell: PropTypes.oneOfType([
+        PropTypes.oneOf([null]),
+        PropTypes.shape({
+          colIndex: PropTypes.number.isRequired,
+          rowIndex: PropTypes.number.isRequired,
+        }),
+      ]).isRequired,
+      isMultipleKeyPressed: PropTypes.bool.isRequired,
+    }).isRequired,
+    options: PropTypes.object.isRequired,
+    pagination: PropTypes.shape({
+      page: PropTypes.number.isRequired,
+      pageCount: PropTypes.number.isRequired,
+      pageSize: PropTypes.number.isRequired,
+      paginationMode: PropTypes.oneOf(['client', 'server']).isRequired,
+      rowCount: PropTypes.number.isRequired,
+    }).isRequired,
+    preferencePanel: PropTypes.shape({
+      open: PropTypes.bool.isRequired,
+      openedPanelValue: PropTypes.oneOf(['columns', 'filters']),
+    }).isRequired,
+    rendering: PropTypes.shape({
+      realScroll: PropTypes.shape({
+        left: PropTypes.number.isRequired,
+        top: PropTypes.number.isRequired,
+      }).isRequired,
+      renderContext: PropTypes.oneOfType([
+        PropTypes.oneOf([null]),
+        PropTypes.shape({
+          /**
+           * The column index of the first rendered column.
+           */
+          firstColIdx: PropTypes.number,
+          /**
+           * The first rendered row in the rendering zone
+           */
+          firstRowIdx: PropTypes.number,
+          /**
+           * The column index of the last rendered column.
+           */
+          lastColIdx: PropTypes.number,
+          /**
+           * The last rendered row in the rendering zone
+           */
+          lastRowIdx: PropTypes.number,
+          /**
+           * The left empty width required to position the viewport at the beginning of the first rendered column.
+           */
+          leftEmptyWidth: PropTypes.number,
+          /**
+           * The rendering zone page calculated with the scroll position
+           */
+          page: PropTypes.number,
+          /**
+           * The current page size if pagination is enabled.
```

Comment:
```suggestion
           * Sets the page size if pagination is enabled.
```
(Guessing here...)

Reply:
We can't set it directly as it's the state prop